pkg/ccl/backupccl/backup_test.go

```diff
@@ -9664,6 +9674,37 @@ func TestBackupRestoreSeparateExplicitIsDefault(t *testing.T) {
 
 		sib := fmt.Sprintf("BACKUP DATABASE fkdb INTO LATEST IN %s WITH incremental_location = %s", dest, inc)
 		sqlDB.Exec(t, sib)
+		{
+			// Locality Aware Show Backup validation
+			// TODO (msbutler): move to data driven test after 22.1 backport
+
+			// Assert the localities field populates correctly (not null if backup is locality aware).
+			localities := sqlDB.QueryStr(t,
+				fmt.Sprintf("SELECT locality FROM [SHOW BACKUP FILES FROM LATEST IN %s]", dest))
+			for _, locality := range localities {
+				if len(br.dest) > 1 {
+					require.NotEqual(t, "NULL", locality[0])
```

Comment:
worth checking the actual locality tag values too since it's a static list? We did something below to render `NULL` as `default` so might be nice to check that works as expected. Datadriven will make this so much easier 😭

Reply:
Done.